Source/WebCore/bindings/js/SerializedScriptValue.cpp

```diff
@@ -3378,39 +3373,42 @@ class CloneDeserializer : public CloneBase {
 
     template <typename T> bool readLittleEndian(T& value)
     {
-        if (m_failed || !readLittleEndian(m_ptr, m_end, value)) {
+        if (m_failed || !readLittleEndian(m_data, value)) {
             SERIALIZE_TRACE("FAIL deserialize");
             fail();
             return false;
         }
         return true;
     }
 #if ASSUME_LITTLE_ENDIAN
-    template <typename T> static bool readLittleEndian(const uint8_t*& ptr, const uint8_t* end, T& value)
+    template <typename T> static bool readLittleEndian(std::span<const uint8_t>& span, T& value)
     {
-        if (ptr > end - sizeof(value))
+        if (span.size() < sizeof(value))
             return false;
 
-        if (sizeof(T) == 1)
-            value = *ptr++;
-        else {
-            value = *reinterpret_cast<const T*>(ptr);
-            ptr += sizeof(T);
+        if (sizeof(T) == 1) {
+            value = span[0];
+            span = span.subspan(1);
```

Comment:
Why do we need a special case here for size 1? If we do need it maybe `if constexpr` is better?